ignite/services/network/node.go

```diff
@@ -0,0 +1,87 @@
+package network
+
+import (
+	"context"
+	"encoding/base64"
+
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	spntypes "github.com/tendermint/spn/pkg/types"
+
+	"github.com/ignite-hq/cli/ignite/pkg/cosmosclient"
+)
+
+type (
+	// IBC is node client info.
+	IBC struct {
+		ConsensusState spntypes.ConsensusState
+		ValidatorSet   spntypes.ValidatorSet
+		UnbondingTime  int64
+		Height         uint64
+	}
+
+	// Node is node builder.
+	Node struct {
+		cosmos       CosmosClient
+		stakingQuery stakingtypes.QueryClient
+	}
+)
+
+func NewNode(ctx context.Context, nodeAPI string) (*Node, error) {
+	c, err := cosmosclient.New(ctx, cosmosclient.WithNodeAddress(nodeAPI))
+	if err != nil {
+		return nil, err
+	}
+	return &Node{
+		cosmos:       c,
+		stakingQuery: stakingtypes.NewQueryClient(c.Context()),
```

Comment:
This query client needs to be configurable from outside through Options, like we do in network.New. for easy testing

Reply:
[It's the same logic used for the network](https://github.com/ignite-hq/cli/blob/e0ff072dde9d0fe4ac1d16c32c696f3d18d46a26/ignite/services/network/network.go#L102-L116). And also, the mocks for the StakingQuery are already there. It's already prepared for the unit test. We only need to find a way to mock the `Node` structure with mockery package.